modules/database/database.bal

```diff
@@ -959,44 +957,155 @@ public isolated transactional function createTimelineSubstep(int stepId, Timelin
     );
 }
 
-# Checks if substep is already in database and updates cleared field in case it is. Otherwise calls createTimelineSubstep. 
+# Updates cleared field of timeline substep
 #
 # + stepId - stepId
 # + substep - substep
-# + return - error or ()
+# + return - error in case no update or multiple updated
 public isolated transactional function updateTimelineSubstep(int stepId, TimelineSubstep substep) returns error? {
-    TimelineSubstepSQL[] substeps = check getTimelineSubsteps(stepId);
-    TimelineSubstepSQL[] duplicates;
-    if substep.substepId != () {
-        duplicates = from var tmpSubstep in substeps
-            where tmpSubstep.href == substep.href && tmpSubstep.hrefUi == substep.hrefUi
-                && tmpSubstep.substepId == substep.substepId
-            select tmpSubstep;
-        if duplicates.length() > 0 {
-            var result = check experimentDB->execute(
-                `UPDATE TimelineSubstep SET cleared=${substep.cleared} WHERE stepId=${stepId} AND href=${substep.href} AND hrefUi=${substep.hrefUi} AND substepId=${substep.substepId};`
+    if substep.href == "" {
+        return error("Href cannot be empty!");
+    }
+    sql:ExecutionResult result;
+    if substep.substepId == () {
```

Comment:
In the "second thought" approach, stepNr should be substepNr, right? So we would only have stepId and substepNr and then call a simple UPDATE that sets all previous substeps to true. 
The main problem was to find the correct substep database entry when the only thing passed from the UI that identifies the substep may be `Href`. This is insufficient as multiple substeps may exist with this Href. Would it not make more sense to refactor the substep data model and simply include the substepNr? The plugin runner already has a field for substepNr. This could simply be passed in addition to the other entries to the UI and then the backend does not need to set the values itself and we have a guarantee to target unique substeps with stepId and substepNr.